### PR TITLE
[12.0] [MIG] project_stage_state

### PR DESCRIPTION
**Pull request by anandkansagra**
Issue [Reference](https://github.com/OCA/project/issues/435)

### CODE REVIEW

File: project_stage_state/README.rst [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+#. Save your changes
+#. Go to Project -> Dashboard and click on a project
+#. Click on task in the stage you just edited
+#. Under the "Extra Info" tab, you can see the "State" field for that task
+
+.. image:: https://odoo-community.org/website/image/ir.attachment/5784_f2813bd/datas
+   :alt: Try me on Runbot
+   :target: https://runbot.odoo-community.org/runbot/140/12.0
+
+Known issues / Roadmap
```
Comment (nikul-serpentcs) [inline project_stage_state/README.rst]: @anandkansagra If not use then remove it

File: project_stage_state/models/project_task.py [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+from odoo import fields, models
+
+
+class ProjectTask(models.Model):
+    """Added state in the Project Task."""
+
+    _inherit = 'project.task'
+
+    state = fields.Selection(
+        related='stage_id.state', store=True, readonly=True)
```
Comment (nikul-serpentcs) [inline project_stage_state/models/project_task.py]: @anandkansagra `related` always `readonly=True` in v12

**Review by nikul-serpentcs — COMMENTED** [review]
Improve Code

File: project_stage_state/models/project_task.py (hunk 2 above)
Comment (anandkansagra) [reply project_stage_state/models/project_task.py]: Thanks, Done

File: project_stage_state/README.rst (hunk 1 above)
Comment (anandkansagra) [reply project_stage_state/README.rst]: Thanks, Done

File: project_stage_state/views/project_view.xml [hunk 3]
```diff
@@ -0,0 +1,27 @@
+<?xml version="1.0" encoding="utf-8"?>
+<odoo>
+
```
Comment (alexey-pelykh) [inline project_stage_state/views/project_view.xml]: A license and copyright notice would be great over here as well, though not required

File: project_stage_state/models/project_task_type.py [hunk 4]
```diff
@@ -0,0 +1,20 @@
+# Daniel Reis, 2014
```
Comment (alexey-pelykh) [inline project_stage_state/models/project_task_type.py]: Copyright should be be as per https://github.com/OCA/maintainer-tools/blob/d9e2c3b8dc163c8d0fcdd36278b782074c3cff12/template/module/__manifest__.py#L2

File: project_stage_state/models/__init__.py [hunk 5]
```diff
@@ -0,0 +1,2 @@
+from . import project_task
```
Comment (alexey-pelykh) [inline project_stage_state/models/__init__.py:3]: License notice is needed in `__init__` as well

File: project_stage_state/__manifest__.py [hunk 6]
```diff
@@ -0,0 +1,19 @@
+# GNU Affero General Public License <http://www.gnu.org/licenses/>
+
+{
+    'name': 'Add State field to Project Stages',
+    'version': '12.0.1.0.0',
+    'category': 'Project Management',
+    'summary': 'Restore State attribute removed from Project Stages in 8.0',
+    'author': "Daniel Reis,Odoo Community Association (OCA)",
+    'website': 'https://github.com/OCA/project-service',
```
Comment (alexey-pelykh) [inline project_stage_state/__manifest__.py]: This should be corrected to actual repository of the module

**Review by alexey-pelykh — CHANGES_REQUESTED** [review]

File: project_stage_state/README.rst [hunk 7]
```diff
@@ -0,0 +1,66 @@
+.. image:: https://img.shields.io/badge/licence-AGPL--3-blue.png
```
Comment (alexey-pelykh) [inline project_stage_state/README.rst:1]: Please split readme as per https://github.com/OCA/maintainer-tools/tree/master/template/module/readme

**Review by alexey-pelykh — CHANGES_REQUESTED** [review]

**Comment by pedrobaeza** [comment]
The same applies here about the commit name said on project_task_material_stock.

File: project_stage_state/views/project_view.xml (hunk 3 above)
Comment (anandkansagra) [reply project_stage_state/views/project_view.xml]: @alexey-pelykh , Could you please check it now?

File: project_stage_state/models/project_task_type.py (hunk 4 above)
Comment (anandkansagra) [reply project_stage_state/models/project_task_type.py]: @alexey-pelykh  , Could you please check it now?

File: project_stage_state/models/__init__.py (hunk 5 above)
Comment (anandkansagra) [reply project_stage_state/models/__init__.py:3]: @alexey-pelykh , Could you please check it now?

File: project_stage_state/__manifest__.py (hunk 6 above)
Comment (anandkansagra) [reply project_stage_state/__manifest__.py]: @alexey-pelykh , Could you please check it now?

File: project_stage_state/README.rst (hunk 7 above)
Comment (anandkansagra) [reply project_stage_state/README.rst:1]: @alexey-pelykh , Could you please check it now?

**Comment by pedrobaeza** [comment]
@anandkansagra there's no need to reply each of the comments if no other action is required. If you have made all the changes, just drop a general comment saying "Everything changed. Please review again" or similar.

File: project_stage_state/models/__init__.py (hunk 5 above)
Comment (alexey-pelykh) [reply project_stage_state/models/__init__.py:3]: Only license, w/o copyright

**Review by alexey-pelykh — CHANGES_REQUESTED** [review]

**Comment by alexey-pelykh** [comment]
@anandkansagra one more tiny change please 👍

**Comment by anandkansagra** [comment]
> @anandkansagra one more tiny change please

@alexey-pelykh , Done

**Review by alexey-pelykh — APPROVED** [review]

**Review by Change2improve — APPROVED** [review]

**Pull request merged**